src/lib/logical_query_plan/functional_dependency.hpp

```diff
@@ -39,4 +39,33 @@ struct FunctionalDependency {
 
 std::ostream& operator<<(std::ostream& stream, const FunctionalDependency& expression);
 
+/**
+ * @return A merger FD set from the given input @param fds_a and @param fds_b. FDs with the same determinant
+ *         expressions are merged into single objects by merging their dependent expressions.
+ */
+std::vector<FunctionalDependency> merge_fds(const std::vector<FunctionalDependency>& fds_a,
+                                            const std::vector<FunctionalDependency>& fds_b);
```

Comment:
I am not sure, where to put this helper. Some other FD-related helper functions are in `lqp_utils.cpp`. However, since this merge function does not involve any LQPs, putting it into `lqp_utils.cpp` also seems wrong.

Reply:
This is fine. In the long term, I wonder if the FD container should become its own class.